crates/memory/src/op_store.rs

```diff
@@ -53,33 +56,63 @@ impl OpStore for Kitsune2MemoryOpStore {
         .boxed()
     }
 
+    /// Store the combined hash of a time slice.
+    ///
+    /// The `slice_id` is the index of the time slice. This is a 0-based index. So for a given
+    /// time period being used to slice time, the first `slice_hash` at `slice_id` 0 would
+    /// represent the combined hash of all known ops in the time slice `[0, period)`. Then `slice_id`
+    /// 1 would represent the combined hash of all known ops in the time slice `[period, 2*period)`.
     fn store_slice_hash(
         &self,
         slice_id: u64,
-        slice_hash: Vec<u8>,
+        slice_hash: bytes::Bytes,
     ) -> BoxFuture<'_, K2Result<()>> {
         async move {
             self.write()
                 .await
                 .time_slice_hashes
-                .insert(slice_id, slice_hash);
-            Ok(())
+                .insert(slice_id, slice_hash)
         }
         .boxed()
     }
 
+    /// Retrieve the count of time slice hashes stored.
+    ///
+    /// Note that this is not the number of hashes that have been provided at unique `slice_id`s.
+    /// Start from time slice id 0 and count up to the highest stored id. This value should be the
+    /// count based on the highest stored id.
+    /// This value is easier to compare between peers because it ignores sync progress. A simple
+    /// count cannot tell the difference between a peer that has synced the first 4 time slices,
+    /// and a peer who has synced the first 3 time slices and created one recent one. However,
+    /// using the highest stored id shows the difference to be 4 and say 300 respectively.
```

Comment:
I would leave it at that. Here's a suggestion that I find clearer:
```suggestion
    /// Note that this is not the total number of hashes of a time slice at a unique `slice_id`.
    /// This value is the highest stored id, starting from time slice id 0 and counting up to the highest stored id. In other words it is the id of the most recent time slice.
    ///
    /// This value is easier to compare between peers because it ignores sync progress. A simple
    /// count cannot tell the difference between a peer that has synced the first 4 time slices,
    /// and a peer who has synced the first 3 time slices and created one recent one. However,
    /// using the highest stored id shows the difference to be 4 and say 300 respectively.
```

Reply:
Okay, happy with most of that, except it does return a genuine count rather than an the id so I've adjusted that a tiny bit further